modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -1260,8 +1250,14 @@ public void update(final WorkflowInstance workflowInstance) throws WorkflowExcep
         // updates for running operations since we updated the metadata right before these operations and will do so
         // again right after those operations.
         if (op == null || op.getState() != OperationState.RUNNING) {
-          updateWorkflowInstanceInIndex(workflowInstance, accessControlList, episodeDublinCoreCatalog,
-                  elasticsearchIndex);
+          // Collect necessary information only for index update
+          long id = workflowInstance.getId();
+          int state = workflowInstance.getState().ordinal();
+          String mpId = workflowInstance.getMediaPackage().getIdentifier().toString();
+          String orgId = workflowInstance.getOrganizationId().toString();
```

Comment:
Isn't this a string already?
```suggestion
          String orgId = workflowInstance.getOrganizationId();
```

Reply:
Yes, it is. This type conversion was necessary in an earlier version of the code and should be removed now.